google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -388,6 +388,9 @@ private static boolean isWindows() {
 
   protected static String getAppEngineProjectId() {
```

Comment:
@neozwu Correct me if I'm wrong. I think some of these checks also work on GCE and GKE right?